lib/src/main/java/com/metaplex/lib/experimental/serialization/format/Borsh.kt

```diff
@@ -0,0 +1,100 @@
+@file:OptIn(ExperimentalSerializationApi::class)
+
+package com.metaplex.lib.experimental.serialization.format
+
+import kotlinx.serialization.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.encoding.*
+import kotlinx.serialization.modules.EmptySerializersModule
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.nio.charset.StandardCharsets
+
+sealed class Borsh : BinaryFormat {
+
+    companion object Default : Borsh()
+
+    override val serializersModule = EmptySerializersModule
+
+    override fun <T> decodeFromByteArray(deserializer: DeserializationStrategy<T>,
+                                         bytes: ByteArray): T =
+        BorshDecoder(bytes).decodeSerializableValue(deserializer)
+
+    override fun <T> encodeToByteArray(serializer: SerializationStrategy<T>, value: T): ByteArray =
+        BorshEncoder().apply { encodeSerializableValue(serializer, value) }.borshEncodedBytes
+}
+
+class BorshDecoder(val bytes: ByteArray) : AbstractDecoder() {
+
+    private val byteBuffer = ByteBuffer.wrap(bytes).apply {
+        order(ByteOrder.LITTLE_ENDIAN)
+        getLong() // consume the account discriminator (we don't need it)
+    }
+
+    override val serializersModule = EmptySerializersModule
+
+    // Not called for sequential decoders
+    override fun decodeElementIndex(descriptor: SerialDescriptor): Int = 0
+    override fun decodeSequentially(): Boolean = true
+    override fun decodeNotNullMark(): Boolean = decodeBoolean()
+
+    override fun decodeCollectionSize(descriptor: SerialDescriptor): Int = decodeInt()
+    override fun decodeEnum(enumDescriptor: SerialDescriptor): Int = decodeInt()
+
+    override fun decodeBoolean(): Boolean = byteBuffer.get().toInt() != 0
+    override fun decodeByte(): Byte = byteBuffer.get()
+    override fun decodeShort(): Short = byteBuffer.short
+    override fun decodeInt(): Int = byteBuffer.int
+    override fun decodeLong(): Long = byteBuffer.long
+    override fun decodeFloat(): Float = byteBuffer.float
+    override fun decodeDouble(): Double = byteBuffer.double
+    override fun decodeChar(): Char = byteBuffer.char
+    override fun decodeString(): String {
+        val length = byteBuffer.int
+        val bytes = ByteArray(length)
+        byteBuffer.get(bytes)
+        return String(bytes, StandardCharsets.UTF_8).replace("\u0000", "")
+    }
+}
+
+class BorshEncoder : AbstractEncoder() {
+    // TODO: need to figure out how to compute the account discriminator,
+    //  for now we are just putting all zeros for the discriminator
+    private val bytes = mutableListOf<Byte>(0, 0, 0, 0, 0, 0, 0, 0)
```

Comment:
the account discriminator is a Solana/Anchor thing, it is not part of the Borsh spec. Remove this and from the Borsh encoder/decoder and instead move it to its own abstraction that can be used on top of borsh